### PR TITLE
Dbz 3085

### PR DESCRIPTION
**Pull request by gunnarmorling**

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Hey @jpechane, can you check out this one? It doesn't compile yet, but it shows the general approach. WDYT?

File: debezium-core/src/main/java/io/debezium/pipeline/DataChangeEvent.java [hunk 1]
```diff
@@ -5,22 +5,13 @@
...
-    public SourceRecord getRecord() {
-        return record;
-    }
+/**
+ * Intermediary representation of data change events, as exchanged between producer threads (e.g. reading the MySQL binlog) and consumer threads (e.g. the Kafka Connect polling loop).
+ *
+ * @author Gunnar Morling
+ *
+ */
+public interface DataChangeEvent {
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/pipeline/DataChangeEvent.java:14]: @gunnarmorling Over I like the solution but this interface seems to be a little bit anemic and I'd say the naming currently does not represent the trait that it provides.
How about making the interface generic and add method like
```
T getData()
```
?

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/pipeline/DataChangeEvent.java:14]: @jpechane, agreed in general; OTOH, there's no real advantage or need for adding that method right now. I'd say let's do this once actually needed.

**Comment by gunnarmorling** [comment]
@jpechane, I don't quite remember -- were there any changes you still wanted to do to the actual size guestimate implementation?

**Comment by jpechane** [comment]
@gunnarmorling `ARRAY` and `MAP` are unimplemented yet

**Comment by gunnarmorling** [comment]
@jpechane, could you push a commit for adding this?

**Comment by gunnarmorling** [comment]
@jpechane, do you still have the code handy which you used for creating the comparison of old and new approach?

**Comment by gunnarmorling** [comment]
@jpechane gentle ping re the last two comments. Thx!

**Comment by jpechane** [comment]
@gunnarmorling The missing code in question is still in this PR, you've removed it in one of the commits :-)
Support for {{ARRAY}} and {{MAP}} is still missing, this will need slight refactoring of the code

**Comment by gunnarmorling** [comment]
> Support for {{ARRAY}} and {{MAP}} is still missing, this will need slight
refactoring of the code

Ok, are you going to push a commit for this, or should I?

>

**Comment by jpechane** [comment]
@gunnarmorling I am on it

**Comment by jpechane** [comment]
@gunnarmorling On question though, it seems to me that the test take much longer than usual. Why is that?

**Comment by jpechane** [comment]
@gunnarmorling Added commit with array/map

**Comment by gunnarmorling** [comment]
> On question though, it seems to me that the test take much longer than usual. Why is that?

@jpechane, hum, that's interesting. Do you have any specific example?

**Comment by jpechane** [comment]
@gunnarmorling Please compare https://github.com/debezium/debezium/runs/2875814390 - 20 mins with https://github.com/debezium/debezium/pull/2332/checks?check_run_id=2882452850 - 75 mins and failure

**Comment by jpechane** [comment]
Partially replaced with https://github.com/debezium/debezium/pull/3707
Additional refactoring related to binding data/size could be done in a separate PR.

**Pull request closed without merging**